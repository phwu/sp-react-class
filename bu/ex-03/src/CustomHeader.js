import React from 'react';

const Header = (props) => {
    return (
        <div>
        <h1>Express Scripts Banking Services</h1>
        <p>Serving the Express Scripts community since {props.today}</p>
        </div>
    );
};

export default Header;