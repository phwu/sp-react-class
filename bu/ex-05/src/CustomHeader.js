import React from 'react';

const CustomHeader = ( props ) => {
  return (
    <header className="page-header">
      {props.children}
    </header>
  );
}

export default CustomHeader;
