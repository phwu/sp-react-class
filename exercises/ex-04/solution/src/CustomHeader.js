import React from 'react';

const CustomHeader = ( props ) => {
  return (
    <header className="page-header">
      <h1>Super-Fake Banking Services</h1>
      <div>Serving the community since {props.today}</div>
      <hr/>
    </header>
  );
};

export default CustomHeader;
