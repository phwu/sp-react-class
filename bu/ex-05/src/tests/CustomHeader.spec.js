import React from 'react';
import renderer from 'react-test-renderer';

import CustomHeader from '../CustomHeader';

it('non snap test', () => {
    expect( 2+2 ).toBe(4);
});

it('Testing snapshot', () => {
    const snap = renderer.create(
        <CustomHeader>lol</CustomHeader>
    ).toJSON();

    expect(snap).toMatchSnapshot();
});
