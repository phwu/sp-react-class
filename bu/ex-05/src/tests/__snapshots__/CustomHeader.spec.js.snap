// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot 1`] = `
<header
  className="page-header"
>
  lol
</header>
`;
